.github/workflows/zenodo-cache-sync.yml

```diff
@@ -0,0 +1,83 @@
+---
+name: zenodo-cache-sync
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - zenodo-cache-action # For testing purposes
+  schedule:
+    - cron: "0 6 * * 1-5" # Weekdays at midnight on MST
+
+env:
+  INTERNAL_ZENODO_CACHE_BUCKET: gs://internal-zenodo-cache.catalyst.coop
+  PUBLIC_ZENODO_CACHE_BUCKET: gs://zenodo-cache.catalyst.coop
+
+jobs:
+  zenodo-cache-sync:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Checkout dev branch
+        uses: actions/checkout@v3
+        with:
+          ref: dev
+
+      - name: Install Conda environment using mamba
+        uses: mamba-org/provision-with-micromamba@v13
+        with:
+          environment-file: test/test-environment.yml
+          cache-env: true
+          channels: conda-forge,defaults
+          channel-priority: strict
+
+      - name: Log environment details
+        run: |
+          conda info
+          conda list
+          conda config --show-sources
+          conda config --show
+          printenv | sort
+
+      - name: Set default gcp credentials
+        id: gcloud-auth
+        uses: "google-github-actions/auth@v0"
+        with:
+          credentials_json: "${{ secrets.ZENODO_CACHE_MANAGER_SA_KEY }}"
+
+      - name: Update GCS cache with any new Zenodo archives
+        run: |
+          pudl_datastore --gcs-cache-path ${{ env.INTERNAL_ZENODO_CACHE_BUCKET }} --bypass-local-cache --loglevel DEBUG
```

Comment:
I think you gotta `pip install --editable ./` before this will work.

Reply:
With `provision-with-micromamba` I don't think you should need to activate the environment -- it should be the system-wide environment. But `pudl` isn't installed in that environment (in the main CI Tox does the installing)